evm/finality-block.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const {
+    providers: { JsonRpcProvider },
+} = ethers;
+const { Command, Option } = require('commander');
+const { printInfo, mainProcessor } = require('./utils');
+const { addBaseOptions } = require('./cli-utils');
+
+async function processCommand(config, chain, options) {
+    let { rpc, waitTime, update } = options;
+
+    if (!rpc) {
+        rpc = chain.rpc;
+    }
+
+    let max = 0;
+    let sum = 0;
+    const DELAY = 10000;
+
+    const provider = new JsonRpcProvider(rpc);
+
+    try {
+        await provider.getBlock('finalized');
+    } catch {
+        if (update && waitTime) {
+            chain.finality = parseInt(waitTime);
+        }
+
+        printInfo('Finalized tag not supported by rpc');
+        return;
+    }
+
+    for (let i = 0; i < 10; i++) {
```

Comment:
Why is 10 chosen here?

Reply:
make it into a configurable value. Default should be higher. 10 queries every 10s won't give you the avg worst case time. Ethereum finality takes 15-20 mins, so the min/max could be achieved anytime during that usually, even longer in some L2s due to slow batching.

Reply:
added as option to be provided as param and default value will be 200, approximating the script to run for 30 mins.